src/CalcManager/UnitConverter.cpp

```diff
@@ -879,16 +879,26 @@ void UnitConverter::Calculate()
         else
         {
             returnValue = stod(m_returnDisplay);
-            wstring returnString;
-            if (m_currentDisplay.size() <= OPTIMALDIGITSALLOWED && abs(returnValue) >= OPTIMALDECIMALALLOWED)
+
+            auto currentDisplayTrimmed = m_currentDisplay;
+            TrimString(currentDisplayTrimmed);
+            int currentNumberSignificantDigits = static_cast<int>(currentDisplayTrimmed.size());
+            if (currentDisplayTrimmed.find(L'.') != currentDisplayTrimmed.npos)
```

Comment:
Doesn't the '.' character change based on the display language or does that happen later?

Reply:
This is safe, this string is locale-independent